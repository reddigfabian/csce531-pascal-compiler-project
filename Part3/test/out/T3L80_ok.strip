b_global_decl (C, alignment = 4, size = 4)
b_global_decl (B, alignment = 4, size = 4)
b_global_decl (A, alignment = 4, size = 4)
b_global_decl (X, alignment = 4, size = 4)
b_global_decl (Y, alignment = 1, size = 1)
b_global_decl (Z, alignment = 4, size = 4)
b_global_decl (W, alignment = 8, size = 8)
b_global_decl (K, alignment = 4, size = 4)
b_global_decl (J, alignment = 4, size = 4)
b_global_decl (I, alignment = 4, size = 4)
b_global_decl (Ca, alignment = 1, size = 8)
b_global_decl (Ia, alignment = 4, size = 24)
b_global_decl (Ra, alignment = 8, size = 24)
b_global_decl (Iaa, alignment = 4, size = 484)
b_global_decl (Iaaa, alignment = 4, size = 240)
b_func_prologue (main)
b_push_ext_addr (X)
b_push_const_int (9)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (X)
b_deref (signed long int)
b_push_const_int (0)
b_arith_rel_op ( >= , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC0)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_x, void)
b_push_ext_addr (X)
b_push_ext_addr (X)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( - , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC1 )
b_push_ext_addr (A)
b_push_const_int (-1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (B)
b_push_const_int (0)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (C)
b_push_const_int (1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_deref (signed long int)
b_push_ext_addr (B)
b_deref (signed long int)
b_arith_rel_op ( > , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC2)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_oops, void)
b_jump ( destination = .LC3 )
b_push_ext_addr (B)
b_deref (signed long int)
b_push_ext_addr (C)
b_deref (signed long int)
b_push_ext_addr (A)
b_deref (signed long int)
b_arith_rel_op ( + , signed long int)
b_arith_rel_op ( == , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC4)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_7, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_1, void)
b_push_const_int (1)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC5)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_2, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_newline, void)
b_jump ( destination = .LC6 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_oops, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_is, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_grooviest, void)
b_push_ext_addr (A)
b_push_const_int (-1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (3)
b_arith_rel_op ( > , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC7)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_oops, void)
b_jump ( destination = .LC8 )
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (3)
b_arith_rel_op ( < , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC9)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_outer, void)
b_push_ext_addr (B)
b_push_const_int (0)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (B)
b_deref (signed long int)
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_arith_rel_op ( < , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC11)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_inner, void)
b_push_ext_addr (C)
b_push_const_int (0)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (C)
b_deref (signed long int)
b_push_const_int (5)
b_arith_rel_op ( < , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC13)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_inner_inner, void)
b_push_ext_addr (C)
b_push_ext_addr (C)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC14 )
b_push_ext_addr (B)
b_push_ext_addr (B)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC12 )
b_push_ext_addr (A)
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC10 )
b_push_ext_addr (X)
b_push_const_int (9)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (X)
b_deref (signed long int)
b_push_const_int (0)
b_arith_rel_op ( >= , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC15)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_looping, void)
b_push_ext_addr (X)
b_push_ext_addr (X)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( - , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC16 )
b_push_ext_addr (X)
b_deref (signed long int)
b_push_const_int (0)
b_arith_rel_op ( != , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC17)
b_push_ext_addr (X)
b_push_const_int (0)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (X)
b_push_const_int (3)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (X)
b_deref (signed long int)
b_push_const_int (0)
b_arith_rel_op ( >= , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC18)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_looping, void)
b_push_ext_addr (B)
b_push_const_int (0)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (B)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( <= , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC20)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_looping_looping, void)
b_push_ext_addr (C)
b_push_const_int (8)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (C)
b_deref (signed long int)
b_push_const_int (10)
b_arith_rel_op ( <= , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC22)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_looping_looping_looping, void)
b_push_ext_addr (C)
b_push_ext_addr (C)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC23 )
b_push_ext_addr (B)
b_push_ext_addr (B)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC21 )
b_push_ext_addr (X)
b_push_ext_addr (X)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( - , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC19 )
b_push_ext_addr (A)
b_push_const_int (1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (B)
b_push_const_int (1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (C)
b_push_const_int (1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (B)
b_deref (signed long int)
b_push_const_int (99)
b_arith_rel_op ( < , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC24)
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (19)
b_arith_rel_op ( > , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC26)
b_push_ext_addr (B)
b_push_ext_addr (B)
b_deref (signed long int)
b_push_const_int (2)
b_arith_rel_op ( * , signed long int)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (3)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC27 )
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (9)
b_arith_rel_op ( > , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC28)
b_push_ext_addr (B)
b_push_ext_addr (B)
b_deref (signed long int)
b_push_const_int (5)
b_arith_rel_op ( - , signed long int)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC29 )
b_push_ext_addr (B)
b_push_ext_addr (B)
b_deref (signed long int)
b_push_const_int (3)
b_arith_rel_op ( * , signed long int)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (2)
b_arith_rel_op ( * , signed long int)
b_assign (signed long int)
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_abc, void)
b_push_ext_addr (C)
b_push_ext_addr (C)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC25 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_newline, void)
b_push_ext_addr (Y)
b_push_const_int (97)
b_convert (signed long int -> unsigned char)
b_assign (unsigned char)
b_pop ()
b_push_ext_addr (I)
b_push_const_int (1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (9)
b_arith_rel_op ( != , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC30)
b_push_ext_addr (Ca)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 1)
b_push_ext_addr (Y)
b_deref (unsigned char)
b_assign (unsigned char)
b_pop ()
b_push_ext_addr (Y)
b_push_ext_addr (Y)
b_deref (unsigned char)
b_convert (unsigned char -> signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_convert (signed long int -> unsigned char)
b_convert (unsigned char -> signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_convert (signed long int -> unsigned char)
b_assign (unsigned char)
b_pop ()
b_push_ext_addr (I)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC31 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_ca, void)
b_push_ext_addr (Ca)
b_push_const_int (3)
b_push_const_int (1)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 1)
b_push_ext_addr (Ca)
b_push_const_int (5)
b_push_const_int (1)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 1)
b_deref (unsigned char)
b_assign (unsigned char)
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_ca, void)
b_push_ext_addr (Y)
b_push_ext_addr (Ca)
b_push_const_int (4)
b_push_const_int (1)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 1)
b_deref (unsigned char)
b_assign (unsigned char)
b_pop ()
b_push_ext_addr (Ca)
b_push_const_int (8)
b_push_const_int (1)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 1)
b_push_ext_addr (Y)
b_deref (unsigned char)
b_assign (unsigned char)
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_ca, void)
b_push_ext_addr (I)
b_push_const_int (1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (7)
b_arith_rel_op ( < , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC32)
b_push_ext_addr (Ia)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (4)
b_arith_rel_op ( + , signed long int)
b_push_const_int (5)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 4)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (3)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (I)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC33 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_ia, void)
b_push_ext_addr (Ia)
b_push_ext_addr (Ia)
b_push_const_int (5)
b_push_const_int (5)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 4)
b_deref (signed long int)
b_push_const_int (6)
b_arith_rel_op ( + , signed long int)
b_push_const_int (5)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 4)
b_push_ext_addr (Ia)
b_push_ext_addr (Ia)
b_push_ext_addr (Ia)
b_push_ext_addr (Ia)
b_push_ext_addr (Ia)
b_push_ext_addr (Ia)
b_push_const_int (10)
b_push_const_int (5)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 4)
b_deref (signed long int)
b_push_const_int (5)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 4)
b_deref (signed long int)
b_push_const_int (5)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 4)
b_deref (signed long int)
b_push_const_int (5)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 4)
b_deref (signed long int)
b_push_const_int (5)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 4)
b_deref (signed long int)
b_push_const_int (5)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 4)
b_deref (signed long int)
b_assign (signed long int)
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_ia, void)
b_push_ext_addr (I)
b_push_const_int (7)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (9)
b_arith_rel_op ( <= , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC34)
b_push_ext_addr (Ra)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (7)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 8)
b_push_ext_addr (Ia)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (5)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 4)
b_deref (signed long int)
b_convert (signed long int -> double)
b_push_const_double (1.8999999999999999e+00)
b_arith_rel_op ( + , double)
b_assign (double)
b_pop ()
b_push_ext_addr (I)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC35 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_ra, void)
b_push_ext_addr (I)
b_push_const_int (7)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (W)
b_push_const_double (1.0000000000000000e+00)
b_assign (double)
b_pop ()
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (9)
b_arith_rel_op ( <= , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC38)
b_push_ext_addr (W)
b_push_ext_addr (W)
b_deref (double)
b_push_ext_addr (Ra)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (7)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 8)
b_deref (double)
b_arith_rel_op ( * , double)
b_assign (double)
b_pop ()
b_push_ext_addr (I)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC39 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_w, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_newline, void)
b_push_ext_addr (J)
b_push_const_int (0)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (J)
b_deref (signed long int)
b_push_const_int (10)
b_arith_rel_op ( <= , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC40)
b_push_ext_addr (Iaa)
b_push_const_int (0)
b_push_const_int (0)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 44)
b_push_ext_addr (J)
b_deref (signed long int)
b_push_const_int (0)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 4)
b_push_const_int (1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (J)
b_push_ext_addr (J)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC41 )
b_push_ext_addr (I)
b_push_const_int (1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (10)
b_arith_rel_op ( <= , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC42)
b_push_ext_addr (Iaa)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (0)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 44)
b_push_const_int (0)
b_push_const_int (0)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 4)
b_push_const_int (1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (J)
b_push_const_int (1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (J)
b_deref (signed long int)
b_push_const_int (10)
b_arith_rel_op ( <= , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC44)
b_push_ext_addr (Iaa)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (0)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 44)
b_push_ext_addr (J)
b_deref (signed long int)
b_push_const_int (0)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 4)
b_push_ext_addr (Iaa)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (0)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 44)
b_push_ext_addr (J)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( - , signed long int)
b_push_const_int (0)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 4)
b_deref (signed long int)
b_push_ext_addr (Iaa)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( - , signed long int)
b_push_const_int (0)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 44)
b_push_ext_addr (J)
b_deref (signed long int)
b_push_const_int (0)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 4)
b_deref (signed long int)
b_arith_rel_op ( + , signed long int)
b_push_const_int (10000)
b_arith_rel_op ( % , signed long int)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (J)
b_push_ext_addr (J)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC45 )
b_push_ext_addr (I)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC43 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_iaa, void)
b_push_ext_addr (I)
b_push_const_int (0)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (2)
b_arith_rel_op ( <= , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC46)
b_push_ext_addr (J)
b_push_const_int (2)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (J)
b_deref (signed long int)
b_push_const_int (5)
b_arith_rel_op ( <= , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC48)
b_push_ext_addr (K)
b_push_const_int (5)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (K)
b_deref (signed long int)
b_push_const_int (9)
b_arith_rel_op ( <= , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC50)
b_push_ext_addr (Iaaa)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (0)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 80)
b_push_ext_addr (J)
b_deref (signed long int)
b_push_const_int (2)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 20)
b_push_ext_addr (K)
b_deref (signed long int)
b_push_const_int (5)
b_arith_rel_op ( - , signed long int)
b_ptr_arith_op ( + , signed long int, size = 4)
b_push_const_int (3)
b_push_const_int (2)
b_push_ext_addr (I)
b_deref (signed long int)
b_arith_rel_op ( * , signed long int)
b_push_ext_addr (J)
b_deref (signed long int)
b_arith_rel_op ( + , signed long int)
b_arith_rel_op ( * , signed long int)
b_push_ext_addr (K)
b_deref (signed long int)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (K)
b_push_ext_addr (K)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC51 )
b_push_ext_addr (J)
b_push_ext_addr (J)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC49 )
b_push_ext_addr (I)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC47 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_iaaa, void)
b_func_epilogue (main)
