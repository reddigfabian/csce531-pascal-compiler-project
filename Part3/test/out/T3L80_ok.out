x = 9
x = 8
x = 7
x = 6
x = 5
x = 4
x = 3
x = 2
x = 1
x = 0
712
 is grooviest
OUTER: a = -1
OUTER: a = 0
INNER: b = 0
INNER INNER: c = 0
INNER INNER: c = 1
INNER INNER: c = 2
INNER INNER: c = 3
INNER INNER: c = 4
OUTER: a = 1
INNER: b = 0
INNER INNER: c = 0
INNER INNER: c = 1
INNER INNER: c = 2
INNER INNER: c = 3
INNER INNER: c = 4
INNER: b = 1
INNER INNER: c = 0
INNER INNER: c = 1
INNER INNER: c = 2
INNER INNER: c = 3
INNER INNER: c = 4
OUTER: a = 2
INNER: b = 0
INNER INNER: c = 0
INNER INNER: c = 1
INNER INNER: c = 2
INNER INNER: c = 3
INNER INNER: c = 4
INNER: b = 1
INNER INNER: c = 0
INNER INNER: c = 1
INNER INNER: c = 2
INNER INNER: c = 3
INNER INNER: c = 4
INNER: b = 2
INNER INNER: c = 0
INNER INNER: c = 1
INNER INNER: c = 2
INNER INNER: c = 3
INNER INNER: c = 4
LOOPING: x = 9
LOOPING: x = 8
LOOPING: x = 7
LOOPING: x = 6
LOOPING: x = 5
LOOPING: x = 4
LOOPING: x = 3
LOOPING: x = 2
LOOPING: x = 1
LOOPING: x = 0
LOOPING: x = 3
LOOPING LOOPING: b = 0
LOOPING LOOPING LOOPING: c = 8
LOOPING LOOPING LOOPING: c = 9
LOOPING LOOPING LOOPING: c = 10
LOOPING LOOPING: b = 1
LOOPING LOOPING LOOPING: c = 8
LOOPING LOOPING LOOPING: c = 9
LOOPING LOOPING LOOPING: c = 10
LOOPING: x = 2
LOOPING LOOPING: b = 0
LOOPING LOOPING LOOPING: c = 8
LOOPING LOOPING LOOPING: c = 9
LOOPING LOOPING LOOPING: c = 10
LOOPING LOOPING: b = 1
LOOPING LOOPING LOOPING: c = 8
LOOPING LOOPING LOOPING: c = 9
LOOPING LOOPING LOOPING: c = 10
LOOPING: x = 1
LOOPING LOOPING: b = 0
LOOPING LOOPING LOOPING: c = 8
LOOPING LOOPING LOOPING: c = 9
LOOPING LOOPING LOOPING: c = 10
LOOPING LOOPING: b = 1
LOOPING LOOPING LOOPING: c = 8
LOOPING LOOPING LOOPING: c = 9
LOOPING LOOPING LOOPING: c = 10
LOOPING: x = 0
LOOPING LOOPING: b = 0
LOOPING LOOPING LOOPING: c = 8
LOOPING LOOPING LOOPING: c = 9
LOOPING LOOPING LOOPING: c = 10
LOOPING LOOPING: b = 1
LOOPING LOOPING LOOPING: c = 8
LOOPING LOOPING LOOPING: c = 9
LOOPING LOOPING LOOPING: c = 10
a = 2; b =  3; c = 1
a = 4; b =  9; c = 2
a = 8; b =  27; c = 3
a = 16; b =  81; c = 4
a = 17; b =  76; c = 5
a = 18; b =  71; c = 6
a = 19; b =  66; c = 7
a = 20; b =  61; c = 8
a = 23; b =  122; c = 9

ca = "acegikmo"
ca = "acigikmo"
ca = "acigikmg"
ia = [ 4 5 6 7 8 9 ]
ia = [ 4 5 6 7 8 4 ]
ra = [ 7.900000 8.900000 9.900000 ]
w = 696.069000

iaa =
[
  [    1    1    1    1    1    1    1    1    1    1    1 ]
  [    1    2    3    4    5    6    7    8    9   10   11 ]
  [    1    3    6   10   15   21   28   36   45   55   66 ]
  [    1    4   10   20   35   56   84  120  165  220  286 ]
  [    1    5   15   35   70  126  210  330  495  715 1001 ]
  [    1    6   21   56  126  252  462  792 1287 2002 3003 ]
  [    1    7   28   84  210  462  924 1716 3003 5005 8008 ]
  [    1    8   36  120  330  792 1716 3432 6435 1440 9448 ]
  [    1    9   45  165  495 1287 3003 6435 2870 4310 3758 ]
  [    1   10   55  220  715 2002 5005 1440 4310 8620 2378 ]
  [    1   11   66  286 1001 3003 8008 9448 3758 2378 4756 ]
]
iaaa =
[
  [
    [   11   12   13   14   15 ]
    [   14   15   16   17   18 ]
    [   17   18   19   20   21 ]
    [   20   21   22   23   24 ]
  ]
  [
    [   17   18   19   20   21 ]
    [   20   21   22   23   24 ]
    [   23   24   25   26   27 ]
    [   26   27   28   29   30 ]
  ]
  [
    [   23   24   25   26   27 ]
    [   26   27   28   29   30 ]
    [   29   30   31   32   33 ]
    [   32   33   34   35   36 ]
  ]
]
